rllib/core/models/specs/specs_base.py

```diff
@@ -170,54 +192,23 @@ def validate(self, tensor: TensorType) -> None:
             ValueError: If the shape or dtype of the tensor does not match the
         """
 
-        expected_type = self.get_type()
-        if not isinstance(tensor, expected_type):
-            raise ValueError(_INVALID_TYPE.format(expected_type, type(tensor).__name__))
+        if not isinstance(tensor, self._expected_type):
+            raise ValueError(
+                _INVALID_TYPE.format(self._expected_type, type(tensor).__name__)
+            )
 
-        shape = self.get_shape(tensor)
+        shape = tuple(tensor.shape)
         if len(shape) != len(self._expected_shape):
             raise ValueError(_INVALID_SHAPE.format(self._expected_shape, shape))
 
         for expected_d, actual_d in zip(self._expected_shape, shape):
             if isinstance(expected_d, int) and expected_d != actual_d:
                 raise ValueError(_INVALID_SHAPE.format(self._expected_shape, shape))
 
-        dtype = self.get_dtype(tensor)
```

Comment:
can we still keep these abstractions? we can implement them by default but the base class would still be extendible. So we can have best of both worlds.